src/main/java/frc/robot/subsystems/Wrist.java

```diff
@@ -0,0 +1,171 @@
+
+public class Wrist extends SubsystemBase {
+  private final WPI_TalonFX m_motor;
+  private final PIDController m_pid;
+  private final DutyCycleEncoder m_absEncoder;
+  private final ShuffleboardTab m_wristTab;
+  private boolean m_enabled = true; 
+  private final SingleJointedArmSim m_armSim =
+    new SingleJointedArmSim(
+      WristConstants.kGearBox, 
+      WristConstants.kGearRatio,
+      WristConstants.kMomentOfInertia,
+      WristConstants.kLength,
+      WristConstants.kMinAngleRads,
+      WristConstants.kMaxAngleRads,
+      true,
+      VecBuilder.fill(2*Math.PI/FalconConstants.kResolution)
+      );
+  private final DutyCycleEncoderSim m_encoderSim;
+
+  // Create a Mechanism2d display of an Arm with a fixed ArmTower and moving Arm.
+  private final Mechanism2d m_mech2d = new Mechanism2d(60, 60);
+  private final MechanismRoot2d m_armPivot = m_mech2d.getRoot("ArmPivot", 30, 30);
+  private final MechanismLigament2d m_armTower =
+    m_armPivot.append(new MechanismLigament2d("ArmTower", 30, -90));
+  private final MechanismLigament2d m_arm =
+    m_armPivot.append(
+        new MechanismLigament2d(
+            "Arm",
+            30,
+            Units.radiansToDegrees(m_armSim.getAngleRads()),
+            6,
+            new Color8Bit(Color.kYellow)));
+  
+  public Wrist(ShuffleboardTab wristTab) {
+    // configure the motor.
+    m_motor = MotorFactory.createTalonFXSupplyLimit(WristConstants.kMotorID, Constants.kRioCAN, WristConstants.kContinuousCurrentLimit, WristConstants.kPeakCurrentLimit, WristConstants.kPeakCurrentDuration);
+    m_motor.setNeutralMode(WristConstants.kNeutralMode);
+    m_motor.setInverted(WristConstants.kMotorInvert); 
+    
+    //SIM
+    SmartDashboard.putData("Arm Sim", m_mech2d);
+    // configure the encoder
+    m_absEncoder = new DutyCycleEncoder(WristConstants.kAbsEncoderPort); 
+    m_encoderSim = new DutyCycleEncoderSim(m_absEncoder);
+
+    // make the PID controller
+    m_pid = new PIDController(WristConstants.kP, WristConstants.kI, WristConstants.kD);
+    // set the PID controller's tolerance
+    m_pid.setTolerance(WristConstants.kTolerance);
+    // go to the initial position
+    setSetpoint(WristConstants.kStowPos);
+
+    if (Constants.kUseTelemetry) {
+      m_wristTab = wristTab;
+      setupShuffleboardTab();
+    }
+  }
+
+  /**
+   * Set the Wrist's desired position.
+   * @param setpoint the desired arm position (in rotations)
+   */
+  public void setSetpoint(double setpoint) {
+    // set the PID integration error to zero.
+    m_pid.reset();
+    // set the PID desired position
+    m_pid.setSetpoint(setpoint);
+  }
+
+  @Override
+  public void periodic() {
+    if (m_enabled) {
+      // calculate the PID power level
+      double pidPower = m_pid.calculate(getAbsEncoderPos(), MathUtil.clamp(m_pid.getSetpoint(), WristConstants.kMinPos, WristConstants.kMaxPos));
+      if (Constants.kLogging) LogManager.addDouble("Wrist/pidOutput", pidPower);
```

Comment:
why is this here when you have a update logs method that is called later in the method